sql/src/main/java/io/crate/metadata/Schemas.java

```diff
@@ -234,6 +242,43 @@ public boolean tableExists(TableIdent tableIdent) {
         return true;
     }
 
+    public static class SchemaAndIndexName {
+
+        public final String schemaName;
+        public final String tableName;
+
+        private SchemaAndIndexName(String schemaName, String tableName) {
+            this.schemaName = schemaName;
+            this.tableName = tableName;
+        }
+    }
+
+    public static SchemaAndIndexName getSchemaAndTableName(String indexName) {
```

Comment:
What happens if the indexName is in the format of a partitioned table?

How is `SchemaAndIndexName` different to `TableIdent` ?

Reply:
Good question, I didn't change the behavior there. It wouldn't work for partitions. I think we can consolidate these APIs.